native/dispatch.c

```diff
@@ -310,6 +310,7 @@ w32_short_name(JNIEnv* env, jstring str) {
 
 static HANDLE
 w32_find_entry(JNIEnv* env, HANDLE handle, const char* funname) {
+  if (funcname[0] == '#') funname = (const char *)atoi(funname + 1);
```

Comment:
Until someone removes it. The scope of security vulnerabilities caused by this in case that happens are pretty major, it costs nothing at runtime and is basic defensive programming, just my 0.02c.